tensorflow/python/data/benchmarks/benchmark_base.py

```diff
@@ -31,6 +31,49 @@
 class DatasetBenchmarkBase(test.Benchmark):
   """Base class for dataset benchmarks."""
 
+  def run_op_benchmark(self,
+                       op,
+                       iters=1,
+                       warmup=True):
+    """Benchmarks the op.
+
+    Runs the op `iters` times. In each iteration, the benchmark measures
+    the time it takes to go execute the op.
+
+    Args:
+      op: The tf op to benchmark.
+      iters: Number of times to repeat the timing.
+      warmup: If true, warms up the session caches by running an untimed run.
+
+    Returns:
+      A float, representing the per-execution wall time of the op in seconds.
+      This is the median time (with respect to `iters`) it takes for the op
+      to be executed `iters` num of times.
+    """
+
+    if context.executing_eagerly():
```

Comment:
You are introducing duplicate code between `run_benchmark` and `run_op_benchmark`. Instead, factor out the common functionality into helper functions (e.g. a function that runs the eager mode benchmark and a function that runs the graph mode benchmark). Makes sense?

Reply:
Yes, it makes sense. I have made an attempt to refactor the common code into `_run_graph_benchmark` and `_run_eager_benchmark` in `benchmark_base.py` as there is a need for benchmarking `tf.data.Datasets` as well as few `ops` such as `map_defun` in `tensorflow.python.data.experimental.ops.map_defun`. Please check and let me know.
Thanks!

Reply:
updated the PR description as well.